psutil/_pslinux.py

```diff
@@ -883,6 +884,15 @@ def cmdline(self):
         return [x for x in data.split('\x00')]
 
     @wrap_exceptions
+    def environ(self):
+        f = open_binary("/proc/%s/environ" % self.pid)
```

Comment:
Mmmm, I suppose that is true, but in that case also your `convert_string()` function should fail.
I think what we need is this:

```
        kw = dict(errors='replace') if PY3 else dict()
        with open_text("%s/%s/environ" % (get_procfs_path(), self.pid), **kw) as f:
             ...
```